pkg/authenticator/authenticator.go

```diff
@@ -200,16 +200,44 @@ func (auth *Authenticator) IsCertExpired() bool {
 	return currentDate.Add(bufferTime).After(certExpiresOn)
 }
 
-// Authenticate sends Conjur an authenticate request and returns
-// the response data. Also manages state of certificates.
-func (auth *Authenticator) Authenticate() ([]byte, error) {
+// Authenticate sends Conjur an authenticate request and writes the response
+// to the token file (after decrypting it if needed). It also manages state of
+// certificates.
+func (auth *Authenticator) Authenticate() error {
 	log.Info(log.CAKC040, auth.Config.Username)
 
+	err := auth.loginIfNeeded()
+	if err != nil {
+		return err
+	}
+
+	authenticationResponse, err := auth.makeAuthenticationRequest()
+	if err != nil {
+		return err
+	}
+
+	parsedResponse, err := auth.parseAuthenticationResponse(authenticationResponse)
+	if err != nil {
+		return log.RecordedError(log.CAKC020)
```

Comment:
All the other methods called in this function does not log the error returned.
Maybe this log should be moved into the `parseAuthenticationResponse` implementation?

Reply:
nice, thanks